ts/src/index.ts

```diff
@@ -112,27 +108,56 @@ function initConfigLocal(config: Config): ProfilerConfig {
   return mergedConfig;
 }
 
+async function getGcpMetadata(
+  retriesLeft: number,
+  backoffMillis: number
+): Promise<[string | undefined, string | undefined, string | undefined]> {
+  try {
+    return await Promise.all([
+      gcpMetadata.project('project-id'),
```

Comment:
I believe `gcpMetadata.project` does retries underneath the covers already. There is some subtlety/complexity here between retrying on network vs. http errors, and I understand that gcpMetadata already has that logic.

Reply:
On certain environments, the metadata server is not available for some time (order of minutes) after start up.

Perhaps it makes sense to default to 0 retries, but I do think we need retries that are more configurable than gcp-metadata'a retries.

Reply:
ofrobots@ This change was motivated by the network delay in envoy startup. Envoy may not finish its initialization until a few seconds after the App has already started. Will ```gcpMetadata.project``` eventually fully support retrying for handling such environments? Since other agents will also face the same issue, will ```gcpMetadata.project```  support configurable retries? It will be cleaner and simpler to have it handled in one place than each agent having its own approach for retries.

Reply:
Adding @JustinBeckwith . If there is a problem interacting with the metadata service, it will affect all the other client libraries as well. My preference would be to fix this in a central place. /cc @JustinBeckwith

Reply:
@JustinBeckwith is there a plan for adding retries in the common libraries? And is there any WIP for supporting Istio?